c7n/resources/airflow.py

```diff
@@ -113,3 +115,71 @@ def process_resource_set(self, client, airflow, tags):
 
 ApacheAirflow.filter_registry.register('marked-for-op', TagActionFilter)
 ApacheAirflow.action_registry.register('mark-for-op', TagDelayedAction)
+
+@ApacheAirflow.action_registry.register('update-environment')
+class UpdateApacheAirflowEnvironment(Action):
+    """
+    Action to update an Airflow environment to
+    set the WebserverAccessMode to PRIVATE_ONLY or PUBLIC_ONLY.
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: set-webserver-access-mode
+            resource: airflow
+            actions:
+              - type: update-environment
+                access_mode: PRIVATE_ONLY
+    """
+
+    permissions = ('airflow:UpdateEnvironment',)
+
+    schema = type_schema(
+        'update-environment',
+        access_mode={'type': 'string', 'enum': ['PRIVATE_ONLY', 'PUBLIC_ONLY']},
+        required=['access_mode']
+    )
+    valid_origin_states = ('AVAILABLE', 'UPDATE_FAILED')
+
+    def process(self, resources):
+        resources = self.filter_resources(resources, 'Status', self.valid_origin_states)
+        client = local_session(self.manager.session_factory).client('mwaa')
+        access_mode = self.data.get('access_mode')
```

Comment:
Would it make sense to also look at `access_mode` on the existing resource and avoid an extra api call to update it if it matches the configuration on the policy?

Reply:
Yes, good idea. Will add